cmd/wick/helpers.go

```diff
@@ -124,3 +148,75 @@ func getErrorFromErrorChannel(resC chan error) error {
 	}
 	return nil
 }
+
+func connect(clientInfo *core.ClientInfo, logTime bool, keepalive int) (*client.Client, error) {
+	var session *client.Client
+	var err error
+	var startTime int64
+
+	if logTime {
+		startTime = time.Now().UnixMilli()
+	}
+	switch clientInfo.AuthMethod {
+	case "anonymous":
+		if clientInfo.PrivateKey != "" {
+			return nil, fmt.Errorf("private key not needed for anonymous auth")
+		}
+		if clientInfo.Ticket != "" {
+			return nil, fmt.Errorf("ticket not needed for anonymous auth")
+		}
+		if clientInfo.Secret != "" {
+			return nil, fmt.Errorf("secret not needed for anonymous auth")
+		}
+		session, err = coreConnectAnonymous(clientInfo, keepalive)
+	case "ticket":
+		if clientInfo.Ticket == "" {
+			return nil, fmt.Errorf("must provide ticket when authMethod is ticket")
+		}
+		session, err = coreConnectTicket(clientInfo, keepalive)
+	case "wampcra":
+		if clientInfo.Secret == "" {
+			return nil, fmt.Errorf("must provide secret when authMethod is wampcra")
+		}
+		session, err = coreConnectCRA(clientInfo, keepalive)
+	case "cryptosign":
+		if clientInfo.PrivateKey == "" {
+			return nil, fmt.Errorf("must provide private key when authMethod is cryptosign")
+		}
+		session, err = coreConnectCryptoSign(clientInfo, keepalive)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if logTime {
+		endTime := time.Now().UnixMilli()
+		log.Printf("session joined in %dms\n", endTime-startTime)
+	}
+	return session, err
+}
+
+func getSessions(clientInfo *core.ClientInfo, sessionCount int, concurrency int,
+	logTime bool, keepalive int) ([]*client.Client, error) {
+	var sessions []*client.Client
+	var mutex sync.Mutex
+	var session *client.Client
+	var err error
+	wp := workerpool.New(concurrency)
+	resC := make(chan error, sessionCount)
+	for i := 0; i < sessionCount; i++ {
+		wp.Submit(func() {
+			session, err = connect(clientInfo, logTime, keepalive)
+			mutex.Lock()
+			sessions = append(sessions, session)
+			mutex.Unlock()
+			resC <- err
+		})
+	}
+
+	wp.StopWait()
+	if err = getErrorFromErrorChannel(resC); err != nil {
```

Comment:
Please move closing of the channel from getErrorFromErrorChannel here.
```suggestion
	wp.StopWait()
	close(resC)
	if err = getErrorFromErrorChannel(resC); err != nil {
```

Reply:
Done